eq-author/src/components/ContentPicker/ContentPicker.js

```diff
@@ -18,6 +18,65 @@ const ActionButtons = styled(ButtonGroup)`
   padding: 1em 0;
   flex: 0 0 auto;
 `;
+
+const getDesiredPath = (
+  dataset,
+  selectedId,
+  config,
+  currentIndex,
+  path = []
+) => {
+  let dataIndex;
+  for (
+    dataIndex = 0;
+    dataIndex < dataset[config[currentIndex].childKey].length;
+    dataIndex++
+  ) {
+    if (config[currentIndex + 1].childKey) {
+      const result = getDesiredPath(
+        dataset[config[currentIndex].childKey][dataIndex],
+        selectedId,
+        config,
+        currentIndex + 1,
+        [...path, { [config[currentIndex].childKey]: dataIndex }]
+      );
+      if (result) {
+        return result;
+      }
+    } else if (
+      dataset[config[currentIndex].childKey][dataIndex].id === selectedId
+    ) {
+      return [...path, { [config[currentIndex].childKey]: dataIndex }];
+    }
+  }
+};
+
+const setSelectedElement = (config, selectedId, data) => {
+  const dataPath = flatten(
+    compact(
+      data.map((test, index) => {
+        const result = getDesiredPath(test, selectedId, config, 0);
+        return result ? [index, ...result] : null;
+      })
+    )
+  );
+  return reduce(
+    dataPath,
+    (acc = [], cur, idx) => {
+      if (idx === 0) {
+        acc.push(data[cur]);
+        return acc;
+      } else {
+        const path = Object.keys(cur)[0];
+        const indexAtPath = Object.values(cur)[0];
+        acc.push(acc[idx - 1][path][indexAtPath]);
+        return acc;
+      }
+    },
+    []
+  );
+};
+
```

Comment:
The logic in `getDesiredPath` and `setSelectedElement` is pretty complex (lots of array index calculations going on). It clearly works, but it might be an idea to add some unit tests around it to document the behaviour more than anything else.

Failing that, a useful comment explaining what the expected inputs and outputs of the function will look like might be justified. Future you will thank you for it! 😄